### DISCUSSION TITLE
How to make custom projection handler in HotChocolate?

### DISCUSSION THREAD

**Discussion by ladislavmrnka** [ChilliCream/hotchocolate#5900] [Q&A]
The [documentation](https://chillicream.com/docs/hotchocolate/v13/fetching-data/projections) states:

> Projections operate on IQueryable by default, but it is possible to create custom providers for projections to support a specific database driver.

So how could I create custom handler for projections? I have custom logic to get the data (no IQueryable) and what I really need is access to some abstraction of the request to see what nested objects are in projection to optimize loading of the data.

Is that possible?

**Comment by Glaerferyn** [comment]
Hello, I am also interested in the outcome of this discussion please :)
